zhmcclient_mock/_urihandler.py

```diff
@@ -453,17 +479,37 @@ class GenericGetPropertiesHandler(object):
     Handler class for generic get of resource properties.
     """
 
-    @staticmethod
-    def get(method, hmc, uri, uri_parms, logon_required):
+    # List of supported query parameters for the Get Properties operation
+    # of the resource type using this class.
+    # Must be overridden in the derived resource handler class, if it supports
+    # query parameters.
+    valid_query_parms_get = []
+
+    @classmethod
+    def get(cls, method, hmc, uri, uri_parms, logon_required):
         # pylint: disable=unused-argument
         """Operation: Get <resource> Properties."""
+        # All URI patterns for Get Properties operations must have a last
+        # match group for the query parms.
+        uri, query_parms = parse_query_parms(method, uri)
+        check_invalid_query_parms(
+            method, uri, query_parms, cls.valid_query_parms_get)
+
         try:
             resource = hmc.lookup_by_uri(uri)
         except KeyError:
             new_exc = InvalidResourceError(method, uri)
             new_exc.__cause__ = None
             raise new_exc  # zhmcclient_mock.InvalidResourceError
-        return resource.properties
+
+        subset_pnames = query_parms.get('properties', None)
+        if subset_pnames:
+            subset_pnames = subset_pnames.split(',')
+            ret_props = {}
+            for pname in subset_pnames:
+                ret_props[pname] = resource.properties[pname]
```

Comment:
I assume: if one uses an invalid property name, you end up with a python key error here?
Is there a way to avoid that (if we would want to avoid it)

Reply:
If the property specified in the query parms is invalid, it has been detected before in `check_invalid_query_parms()`.

The other question is whether the resource has the property available. `hmc.lookup_by_uri()` returns the FakedResource, i.e. the internal data representation of the resource. It has all properties that have been defined when the faked resource was created. So there is a chance that the creation of the resource missed the (valid) property. I defined a new exception class `MockedResourceError` for this kind of error and added a check for the presence of the property.